x-pack/plugins/security/public/plugin.tsx

```diff
@@ -207,6 +209,16 @@ export class SecurityPlugin
         suggest: this.securityApiClients.userProfiles.suggest.bind(
           this.securityApiClients.userProfiles
         ),
+        update: this.securityApiClients.userProfiles.update.bind(
+          this.securityApiClients.userProfiles
+        ),
+        userProfile$: this.securityApiClients.userProfiles.userProfile$,
+      },
+      hooks: {
+        useUpdateUserProfile: getUseUpdateUserProfile({
+          apiClient: this.securityApiClients.userProfiles,
+          notifications: core.notifications,
+        }),
```

Comment:
Sorry, I'm only just seeing that you're exporting `useUpdateUserProfile` as part of our plugin contract. React components and hooks should be exported as packages. They do not belong in the plugin contract. I know we have a few legacy components that use this approach but that's because they were introduced before we had the ability to create packages. 

You can use the existing `@kbn/user-profile-components` to export `useUpdateUserProfile` hook. Alternatively you can also move the hook to the `cloud_integrations` plugin but this is not something we want to support from the security plugin contract.

Reply:
Yeah I didn't want to create a package for a hook. Seemed overkilled to me.
The point is to reuse the same logic inside the `security` plugin and the `cloud_integrations` plugin and I don't think the owner of the hook should be the `cloud_integrations` TBH (that the `security` plugin depend on).

Reply:
I will look into the `@kbn/user-profile-components` package.